### ISSUE TITLE
Slower velocity of ball when hit in negative Z direction

### ISSUE THREAD

**Issue by troach06** [TomorrowTodayLabs/NewtonVR#135]
So I'm developing a tennis game for the HTC Vive and I have been using NewtonVR for physics. Everything works smoothly, besides one major bug. Whenever I turn my body around 180 degrees and hit the ball, the ball seems to have less velocity coming off the racket. This is only when I am hitting the ball in the negative Z direction. Weirdly, if I hit a serve, sometimes the ball acts normally. But whenever I do a backhand or forehand it seems dead. Also, when I hit the ball in the negative X direction, it does the same thing and seems to have a slight curve as well. I am not adding any artificial velocity, only angular velocity which I disabled and the bug did not disappear. This is the only thing that is preventing me from putting the app on Steam. If anyone has any idea what could be happening, please let me know. I would appreciate it greatly.

Thanks,
Tyler

**Comment by keithbradner** [comment]
Hey did you get this resolved? This sounds like a very strange issue.

**Comment by troach06** [comment]
Hey, I just saw your reply. Unfortunately, the bug is still there. I have tried basically everything, yet the ball still goes slower in the negative z direction. The weird thing is that it depends on the angle of the collision, as when I do an overhead the ball seems to act normal. The physics work great besides this one glitch, and it is a big one. Have you ever encountered this? I am pretty positive it is a bug with NewtonVR as I have tried literally everything I could think of. Please let me know what I can do to resolve this issue.

**Comment by troach06** [comment]
I ended up fixing the bug. It turns out it was an issue with my code, I had a method where I would detect a collision and then stop detecting collisions for about 0.5 seconds in order to prevent double hits. Somehow this caused the ball to not go as far in the negative Z direction. Once I removed the method all was good and I'm not getting nearly as many double hits as I was in the beginning when I added this method.
